docs/loki-deployment.md

```diff
@@ -0,0 +1,160 @@
+# Overview
+This page lists the 2 common Loki deployments, to help user to better understand Loki architecture.
+
+Loki consists several components as below (Quote from slides [here](https://grafana.com/blog/2018/12/12/loki-prometheus-inspired-open-source-logging-for-cloud-natives/)):
+![Loki Arch](loki_arch.png)
+
+|  Component | Intro |
+| --- | --- |
+| Distributor | use consistent hashing to assign a logstream to an ingester |
+| Ingester | gzip logs and append them. Index and data chunks go to different store |
+| Querier | look at the index to figure out which chunks match, and distributed grep through them. Talks to chunks and Ingester(get the recent data that has not been flushed) both |
+| Table-Manager | manage the chunk and index lifecycle |
+
+
+# Loki Deployment on Single Process
+To those who want to just have a try at Loki, deploying Loki on `Single Process` is the easiest way.
+It combines all above component in a single process.
+
+[Run Locally Using Docker](https://github.com/grafana/loki/tree/master/production#run-locally-using-docker) and [Run from Source](https://github.com/grafana/loki/tree/master/production#build-and-run-from-source) have already gave the examples.
+
+They are using Loki config like below
+```yaml
+auth_enabled: false
+
+server:
+  http_listen_port: 3100
+
+ingester:
+  lifecycler:
+    address: 127.0.0.1
+    ring:
+      kvstore:
+        store: inmemory
+      replication_factor: 1
+  chunk_idle_period: 15m
+
+schema_config:
+  configs:
+  - from: 2018-04-15
+    store: boltdb
+    object_store: filesystem
+    schema: v9
+    index:
+      prefix: index_
+      period: 168h
+
+storage_config:
+  boltdb:
+    directory: /tmp/loki/index
+
+  filesystem:
+    directory: /tmp/loki/chunks
+
+limits_config:
+  enforce_metric_name: false
+  reject_old_samples: true
+  reject_old_samples_max_age: 168h
+
+chunk_store_config:
+  max_look_back_period: 0
+
+table_manager:
+  chunk_tables_provisioning:
+    inactive_read_throughput: 0
+    inactive_write_throughput: 0
+    provisioned_read_throughput: 0
+    provisioned_write_throughput: 0
+  index_tables_provisioning:
+    inactive_read_throughput: 0
+    inactive_write_throughput: 0
+    provisioned_read_throughput: 0
+    provisioned_write_throughput: 0
+  retention_deletes_enabled: false
+  retention_period: 0
+```
+
+# Loki Deployment on Separate Mode
+To those who want to use Loki in production and persuit higher performance, deploying Loki on `Separate Mode` is a must.
+
+There're 4 things need to consider compared to Single Process Mode.
+1. The Ring Store:
+    
+    You need to start a Consul server as the Ring store. It will store a consistent hash ring which consists of a list of tokens and ingesters. And Components like Distributors/Queriers rely on it to know which ingester is alive.
```

Comment:
```suggestion
    You need to start a Consul server as the Ring store. It will store a consistent hash ring which consists of a list of tokens and ingesters. Components like Distributors or Queriers rely on this to know which ingesters are ready.
```